Rectify/run_code.py

```diff
@@ -0,0 +1,47 @@
+import subprocess as sub
+import time
+
+
+def fun(code, user_code, input):
+	"""
+		'input' is user input, 'code' is correct code, and 'user_code' is code submitted by the user
+	"""
+	
+	##code, user_code, input can also be files in which case fileIO is not required
+	input_file = open('input', 'w')
+	input_file.write(input)
+	input_file.close()
+
+	code_file = open('user_code.c', 'w')
+	code_file.write(user_code)
+	code_file.close()
+
+	code_file = open('code.c', 'w')
+	code_file.write(code)
+	code_file.close()
+	##
+
+	not_compiled = sub.call(['gcc', '-o', 'user_code', 'user_code.c'])
+	if not_compiled:
+		return "Compilation Error"
+	sub.call(['gcc', '-o', 'code', 'code.c'])
+
+	process = sub.Popen('./user_code < input >user_out', shell=True)
```

Comment:
Instead of writing input to a file and then piping it - use subprocess PIPE for stdout and stdin